### DISCUSSION TITLE
placing custom block

### DISCUSSION THREAD

**Discussion by Ravensk23** [bridge-core/bridge.#363] [Q&A]
So when i launch my world i can spawn in my item but when i do so and place it it looks like a dirt block with the words update on it anyone know how to fix this?

**Comment by Chikorita-Lover** [comment]
Assuming you mean a custom block by "item," make sure you've set the block's texture via the `blocks.json` and `terrain_texture.json` files. You can learn about this at https://wiki.bedrock.dev/guide/custom-blocks#block-resource-file.

**Reply by Ravensk23** [reply]
ive got the block textures set, could it be anything else?

**Reply by Chikorita-Lover** [reply]
Hmm, could you send the `blocks.json` and `terrain_texture.json` files?

**Reply by Ravensk23** [reply]
![terrain texture json](https://user-images.githubusercontent.com/64230527/121786471-8953ee00-cb85-11eb-8589-ffc44d4298bb.PNG)
![block json](https://user-images.githubusercontent.com/64230527/121786472-8953ee00-cb85-11eb-8f4b-19d6832f207f.png)

**Reply by Chikorita-Lover** [reply]
I would try renaming the texture short name to `tvwastaken` (make sure this is done in both the block behavior and `terrain_texture.json` files).
For the geometry issue mentioned separately, are you sure the identifier isn't `geometry.TvWasTaken` instead of `geometry . TvWasTaken`?

**Reply by Ravensk23** [reply]
i double checked and its was geometry. not geometry . and which one is the texture short name?

**Reply by Chikorita-Lover** [reply]
The texture short name is what's used in `minecraft:material_instances`/`blocks.json` and defined in `terrain_texture.json`/`flipbook_textures.json`.

**Reply by Ravensk23** [reply]
okay ill try that

**Reply by Ravensk23** [reply]
okay so the block is now in game but parts of the model are in parts where they shouldnt be and the model is black and purple

**Reply by Chikorita-Lover** [reply]
Could you send the current block behavior and `terrain_texture.json` files?

**Reply by Ravensk23** [reply]
![Capture2](https://user-images.githubusercontent.com/64230527/121787140-113bf700-cb8a-11eb-9cc2-30064b3b4cae.PNG)
![Capture](https://user-images.githubusercontent.com/64230527/121787141-11d48d80-cb8a-11eb-9ca8-744ef4b10013.PNG)

**Reply by Chikorita-Lover** [reply]
> I would try renaming the texture short name to `tvwastaken` (make sure this is done in both the block behavior and `terrain_texture.json` files).

Make sure `TvWasTaken` in `minecraft:material_instances` is updated to `tvwastaken` as well.

**Reply by Ravensk23** [reply]
hmmm now its telling me that it contains 2 boxes out of bounds and the model does not look how it should even though in bridge when i click entity it has the right shape at least but not in game

![tvhead (1)](https://user-images.githubusercontent.com/64230527/121787553-4ba69380-cb8c-11eb-9b31-760e429feba2.PNG)
![tvheadbridge](https://user-images.githubusercontent.com/64230527/121787556-4e08ed80-cb8c-11eb-8c8d-f41a6346c8d5.PNG)

**Reply by Chikorita-Lover** [reply]
Oh, you have pivot points in your geometry. I've heard before that having pivot points for individual cubes can cause issues, although I'm not 100% sure.
For the texture issue, double check that the texture is at `RP/textures/blocks/TvWasTaken.png`, with the same capitalization and spelling.

**Reply by Ravensk23** [reply]
yeah they have the same capitalization, i think the model it self being thrown out of walk it messing up the texture but my thing is , is that the 2nd picture is from bridge not my model making software so bridge itself sees how the model should be set but for some reason is not putting it in the game that way

**Reply by Chikorita-Lover** [reply]
> Oh, you have pivot points in your geometry. I've heard before that having pivot points for individual cubes can cause issues, although I'm not 100% sure.

I believe this may be why the geometry appears incorrect in-game.

**Reply by Ravensk23** [reply]
hmm and i have no idea if i can fix that

**Reply by Chikorita-Lover** [reply]
If you remove the `minecraft:geometry` block component, does the texture appear as intended?

**Reply by Ravensk23** [reply]
i tried and am now back at the minecraft update dirt block

**Reply by Chikorita-Lover** [reply]
I would try renaming the PNG texture to use no capitals, and update the `terrain_texture.json` file appropriately. A basic rule of thumb is to never use capital letters in add-ons, except for within LANG files and the `minecraft:display_name` component.

**Reply by Ravensk23** [reply]
okay so the texture is on correctly now but the model is still messed up in game but not in bridges model previewer

**Reply by Chikorita-Lover** [reply]
> Oh, you have pivot points in your geometry. I've heard before that having pivot points for individual cubes can cause issues, although I'm not 100% sure.

I would edit the geometry file in something such as VS Code (Notepad would work fine) and remove all the pivot points in the geometry.

**Reply by Ravensk23** [reply]
hmmm i removed the pivot points in the geometry file but it unfortunately  didnt help

**Reply by Chikorita-Lover** [reply]
Does the geometry appear the same in-game?

**Reply by Ravensk23** [reply]
there seems to be some differences trying to fix that now

**Reply by Ravensk23** [reply]
it looks like i have to manually change its those blocks rotation for some reason they dont correspond 
correctly

**Comment by Ravensk23** [comment]
Okay for some reason its now telling me a geometry file doesnt exist
